### ISSUE TITLE
Clean up event listeners in reader_view.js

### ISSUE THREAD

**Issue by einarstubhaug** [readium/readium-shared-js#164]
I've come across an issue with a single-page javascript application.

We instantiate a ReaderView. When the user leaves this view, we remove the Readium iFrame from DOM alltogether (later, if the user wants to go back, we populate it from scratch). 

The issue: If we have left the view and removed the iframe, and we resize the browser window , we get an error message from Readium. We managed to trace it to this line of code where an event listener is attached to the Window:
https://github.com/readium/readium-shared-js/blob/master/js/views/reader_view.js#L67

```
$(window).on("resize.ReadiumSDK.readerView", lazyResize);
```

If we, when leaving the view, remove the event listener like this: 
```
$(window).off("resize.ReadiumSDK.readerView");
```
our code works. But I guess there might be more events being listened to which might cause problems later on.

Are there any ways to reset and remove a ReaderView's event listeners in the Readium API?

Tips much appreciated, 

Einar

**Comment by danielweck** [comment]
Obviously, the catch-all `$window.off()` a a bit brutal so I wouldn't recommend it :)
Take a look at `EpubReader.unloadReaderUI()`:
https://github.com/readium/readium-js-viewer/blob/develop/lib/EpubReader.js#L860
```
var unloadReaderUI = function(){

        if (readium) {
            readium.closePackageDocument();
        }
        
        // needed only if access keys can potentially be used to open a book while a dialog is opened, because keyboard.scope() is not accounted for with HTML access keys :(
        // for example: settings dialogs is open => SHIFT CTRL [B] access key => library view opens with transparent black overlay!
        Dialogs.closeModal();
        Dialogs.reset();
        $('#settings-dialog').modal('hide');
        $('#about-dialog').modal('hide');
        $('.modal-backdrop').remove();
        
        
        Keyboard.off('reader');
        Keyboard.off('settings');

        $('#settings-dialog').off('hidden.bs.modal');
        $('#settings-dialog').off('shown.bs.modal');

        $('#about-dialog').off('hidden.bs.modal');
        $('#about-dialog').off('shown.bs.modal');

        // visibility check fails because iframe is unloaded
        //if (readium.reader.isMediaOverlayAvailable())
        if (readium && readium.reader) // window.push/popstate
            readium.reader.pauseMediaOverlay();
        
        $(window).off('resize');
        $(window).off('mousemove');
        $(window).off('keyup');
        $(window).off('message');
        window.clearTimeout(hideTimeoutId);
        $(document.body).removeClass('embedded');
        document.removeEventListener('click', hideLoop, true);
        $('.book-title-header').remove();

        $(document.body).removeClass('hide-ui');
    }
```

**Comment by jansc** [comment]
I suggest that readium-shared-js and readium-js should provide its own unload function, e.g.

```
ReadiumSDK.reader.unload()
```
and possibly
```
Readium.unload();
```

That way, readium-js-viewer does not have to know about the internals in readium-shared-js. We could contribute pull-requests if you agree, @danielweck.

**Comment by danielweck** [comment]
This sounds great @jansc 
In fact today we have been discussing the implementation proposal to make the three ReadiumJS repositories ; i.e. individual "projects" ; more modular and more flexible (RequireJS + plugins framework, see https://github.com/readium/readium-shared-js/pull/163 ), and it makes a lot of sense to expose a `setup` + `teardown` workflow for each item.

**Comment by jansc** [comment]
Thanks for the info, @danielweck. It sounds like there is no use implementing this in the develop branch with the new feature branch right around the corner. Did you discuss when you plan to merge the plugins branch into develop?

**Comment by danielweck** [comment]
No firm ETA but we are working hard on it, as it is a high priority development task right now. One obvious prerequisite for the develop-merge is to make sure that Readium native application launchers are updated and tested with the new RequireJS architecture. There's a WIP feature/plugins branch in SDKLauncher-OSX which we use as a test bed.

**Comment by jansc** [comment]
I guess I won't be any help regarding the SDK, but drop me a message if there's anything I can assist with in JavaScript-land.
